src/engagement_db_to_analysis/regression_analysis/r_utils.py

```diff
@@ -30,16 +27,12 @@ def convert_dicts_to_r_data_frame(dicts, types):
     lists = defaultdict(list)
     for d in dicts:
         for (k, v) in d.items():
-            lists[k].append(v)
+            lists[k].append(None if v is None else str(v))
 
     # 2. Convert each list of values to an R vector
     vectors = dict()
     for (dataset_name, values) in lists.items():
-        if types[dataset_name] == int:
-            vectors[dataset_name] = IntVector(values)
-        else:
-            assert types[dataset_name] == str, types[dataset_name]
-            vectors[dataset_name] = StrVector(values)
+        vectors[dataset_name] = FactorVector(values)
```

Comment:
Can you add a comment i.e, "Use FactorVector to create factors from categorical data" or 
"Factors are suitable for representing and analyzing categorical variables". it is very clear from the docstrings that `_get_participant_regression_data` returns categorical data. I also think `convert_dicts_to_r_data_frame` should display the same.

Reply:
I have updated the docstring for this function to add 'Each column in the returned data-frame contains a "FactorVector", which is the datatype used to represent categorical data in R.' 

I agree we should make it clear how the data will be interpreted during the conversion. I wonder if we should also rename the function itself to something like "convert_dicts_to_r_data_frame_of_factors" to make it clearer from the name? I can't think of a more succinct way to name it that makes it clear what you get back...

Reply:
sure `convert_dicts_to_r_data_frame_of_factors` is much clearer